src/luascript.cpp

```diff
@@ -4413,12 +4438,76 @@ int LuaScriptInterface::luaGameStartRaid(lua_State* L)
 	const std::string& raidName = getString(L, 1);
 
 	Raid* raid = g_game.raids.getRaidByName(raidName);
-	if (raid) {
-		raid->startRaid();
-		pushBoolean(L, true);
+	if(!raid || !raid->isLoaded()) {
+		lua_pushnumber(L, RETURNVALUE_NOSUCHRAID);
+		return 1;
+	} else if (g_game.raids.getRunning()) {
+		lua_pushnumber(L, RETURNVALUE_ANOTHERRAIDEXECUTING);
+		return 1;
+	}
+
+	RaidEvent* raidEvent = raid->getNextRaidEvent();
+	if (!raidEvent) {
+		lua_pushnumber(L, RETURNVALUE_RAIDCONTAINSNODATA);
+		return 1;
+	}
+
+	g_game.raids.setRunning(raid);
+	raid->startRaid();
+	lua_pushnumber(L, RETURNVALUE_NOERROR);
+	return 1;
+}
+
+int LuaScriptInterface::luaGameReload(lua_State* L)
+{
+	// Game.reload(interfaceName)
+	const std::string& interfaceName = asLowerCaseString(getString(L, 1));
```

Comment:
It won't work, you are binding a rvalue to a reference. The return for `asLowerCaseString(...)` is destroyed as soon as it's assigned, and as it's assigned to a reference, the content of `interfaceName` become invalid.